internal/handler/control.go

```diff
@@ -1,22 +1,142 @@
 // -*- Mode: Go; indent-tabs-mode: t -*-
 //
 // Copyright (C) 2017-2018 Canonical Ltd
+// Copyright (C) 2020 IOTech Ltd
 //
 // SPDX-License-Identifier: Apache-2.0
 
 package handler
 
 import (
+	"context"
 	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+	"sync/atomic"
+	"time"
 
+	"github.com/edgexfoundry/device-sdk-go/internal/cache"
 	"github.com/edgexfoundry/device-sdk-go/internal/common"
+	"github.com/edgexfoundry/device-sdk-go/pkg/models"
+	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
+	"github.com/google/uuid"
 )
 
-func DiscoveryHandler(requestMap map[string]string) {
-	common.LoggingClient.Info(fmt.Sprintf("service: discovery request"))
-}
+var (
+	id     string
+	locker uint32
+)
 
 func TransformHandler(requestMap map[string]string) (map[string]string, common.AppError) {
 	common.LoggingClient.Info(fmt.Sprintf("service: transform request: transformData: %s", requestMap["transformData"]))
 	return requestMap, nil
 }
+
+func DiscoveryHandler(w http.ResponseWriter) {
+	if id == "" {
+		id = uuid.New().String()
+	}
+
+	if w != nil {
+		msg := fmt.Sprintf("Discovery triggered or already running, id = %s", id)
+		w.WriteHeader(http.StatusAccepted) //status=202
```

Comment:
I know the design says 202 should be returned in either case, but I want to make the argument that it might make sense to return separate results so that a client has some idea (a) if a specific request triggered discovery or (b) discovery is busy.

@iain-anderson thoughts? Let's plan to discuss Monday.

Reply:
Could discriminate the two cases by the http return body. The CSDK returns different messages in text/plain. We could design a response object to hold the necessary information but that would be scope creep for this PR. Create a new issue against the v2 API?

Reply:
Let's leave it as is. I checked and there really isn't a decent HTTP status code that equates to "I'm busy". 503 - Server unavailable is probably the closest, but I think we already use that to mean the device service is unavailable. The true way to fix this would be to return a JSON reply message which could embed an application-specific error code.

As for v2? Are we planning on re-implementing this endpoint?